### DISCUSSION TITLE
Do I need to open-source and undergo code review or audit before publishing an snap app developed using MetaMask Snap?

### DISCUSSION THREAD

**Discussion by syd2011** [MetaMask/snaps#1585] [Q&A] [answered]
our group developed a snap app, and published to npm .
Do I need to open-source and undergo code review or audit before publishing an snap app ?

**Comment by Montoya** [comment, marked as answer]
Hi @syd2011! Please have a look at our v1 Readiness Guide, specifically the allowlist process, to see our requirements for our public launch: https://github.com/MetaMask/snaps/discussions/1411

To be clear, you do not need to do any of that to publish your snap to npm. Anyone will be able to install your snap on MetaMask Flask once you publish to npm. But if you want users to be able to install your snap on MetaMask stable when we launch Snaps to our extension later this year, then see the steps in the v1 Readiness Guide.

**Reply by syd2011** [reply]
thanks montoya
